forest/shared/src/cli/snapshot_fetch.rs

```diff
@@ -375,13 +378,20 @@ where
     let mut downloaded: u64 = 0;
     let mut stream = snapshot_response.into_body();
 
-    let mut snapshot_hasher = Sha256::new();
+    let mut snapshot_hasher = if use_compressed {
+        None
```

Comment:
What is the reason behind not checking against checksum in case of compressed snapshot and without aria2? Don't we have a digest in both case? (with and without aria2)

Reply:
Checksum is not provided for a compressed snapshot. Also zstd has checksum mechanism internally

Reply:
Hmm if I look into the digest file I have:
```
cat 468720_2023_04_13T12_13_00Z.sha256sum
ae6d21b3bd88fa0d21528343b05e6f323a7131fad5d3e1148a5ed7c603d915a5 *468720_2023_04_13T12_13_00Z.car
180819491d0dfd37cc09ebc5ceded668c09fa6ec3f1fefe54ad20ca0fa71805a *468720_2023_04_13T12_13_00Z.car.zst
```

But if zstd does it internally, then we can skip it. 👍

Reply:
Did not realize checksum is there. I can investigate adding back checksum check then. Thx for the info

Reply:
👍

Reply:
Just checked the [zstd spec](https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#zstandard-frames) again, its checksum mechanism is optional, we should indeed verify checksum if it's available